### PR TITLE
feat(2399): Remove command tag

### PR DESCRIPTION
**Pull request by ibu1224**
## Context

Create a feature for the following issue
https://github.com/screwdriver-cd/screwdriver/issues/2399

## Objective

In this PR, we will create a new endpoint for removing command tags.
The endpoint is designed in `{namespace}/{name}/tags/{tagName}`.

The command must have a namespace when it publishes.
Unlike template, namespace can be included in the parameters.

## References

- issue: https://github.com/screwdriver-cd/screwdriver/issues/2399
- remove template tag : [code](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/templates/removeTag.js)

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: test/plugins/commands.test.js [hunk 1]
```diff
@@ -1153,6 +1153,71 @@ describe('command plugin test', () => {
...
+
+        it('returns 403 when pipelineId does not match', () => {
+            options.auth.credentials.isPR = true;
+
+            return server.inject(options).then(reply => {
+                assert.equal(reply.statusCode, 403);
+            });
+        });
+
+        it('returns 404 when template tag does not exist', () => {
```
Comment (yoshwata) [inline test/plugins/commands.test.js:1206]: template?

File: test/plugins/commands.test.js [hunk 2]
```diff
@@ -1153,6 +1153,71 @@ describe('command plugin test', () => {
...
+            commandTagFactoryMock.get.resolves(null);
+
+            return server.inject(options).then(reply => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('deletes template tag if has good permission and tag exists', () =>
```
Comment (yoshwata) [inline test/plugins/commands.test.js:1214]: template?

File: test/plugins/commands.test.js (hunk 1 above)
Comment (ibu1224) [reply test/plugins/commands.test.js]: 🙇‍♂️

File: test/plugins/commands.test.js (hunk 2 above)
Comment (ibu1224) [reply test/plugins/commands.test.js]: 🙇‍♂️

**Review by yoshwata — APPROVED** [review]

File: plugins/commands/removeTag.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            strategies: ['token'],
+            scope: ['build']
+        },
+        handler: async (request, h) => {
+            const { pipelineFactory, commandFactory, commandTagFactory } = request.server.app;
+            const { pipelineId, isPR } = request.auth.credentials;
+            const { namespace, name } = request.params;
+            const tag = request.params.tagName;
+
+            console.log(name);
```
Comment (tk3fftk) [inline plugins/commands/removeTag.js:28]: Is this needed? and we should use logger if needed

File: plugins/commands/removeTag.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+/* Currently, only build scope is allowed to tag command due to security reasons.
+ * The same pipeline that publishes the commands has the permission to tag it.
+ */
+
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/commands/{namespace}/{name}/tags/{tagName}',
+    options: {
+        description: 'Delete a command tag',
+        notes: 'Delete a specific commands',
```
Comment (tk3fftk) [inline plugins/commands/removeTag.js:16]:
```suggestion
        notes: 'Delete a specific command tag',
```

File: plugins/commands/removeTag.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            const { namespace, name } = request.params;
+            const tag = request.params.tagName;
+
+            console.log(name);
+
+            return commandTagFactory
+                .get({ namespace, name, tag })
+                .then(commandTag => {
+                    if (!commandTag) {
+                        throw boom.notFound('Commands tag does not exist');
```
Comment (tk3fftk) [inline plugins/commands/removeTag.js:34]: How about little more detailed log?

```suggestion
                        throw boom.notFound(`Command tag ${tag} does not exist for ${namespace}/${name}`);
```

File: plugins/commands/removeTag.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                            namespace,
+                            version: commandTag.version
+                        })
+                    ]).then(([pipeline, command]) => {
+                        // Check for permission
+                        if (pipeline.id !== command.pipelineId || isPR) {
+                            throw boom.forbidden('Not allowed to delete this command tag');
+                        }
+
+                        // Remove the command tag, not the comannd
```
Comment (tk3fftk) [inline plugins/commands/removeTag.js:50]: typo
```suggestion
                        // Remove the command tag, not the command
```

File: plugins/commands/removeTag.js (hunk 3 above)
Comment (ibu1224) [reply plugins/commands/removeTag.js:28]: It's for development. I'll delete it.

File: plugins/commands/removeTag.js (hunk 4 above)
Comment (ibu1224) [reply plugins/commands/removeTag.js]: 🙇‍♂️

File: plugins/commands/removeTag.js (hunk 5 above)
Comment (ibu1224) [reply plugins/commands/removeTag.js]: thank you

**Review by wahapo — APPROVED** [review]

**Review by okuryu — APPROVED** [review]

File: test/plugins/commands.test.js [hunk 7]
```diff
@@ -1153,6 +1153,71 @@ describe('command plugin test', () => {
...
+
+        it('returns 403 when pipelineId does not match', () => {
+            commandMock.pipelineId = 8888;
+
+            return server.inject(options).then(reply => {
+                assert.equal(reply.statusCode, 403);
+            });
+        });
+
+        it('returns 403 when pipelineId does not match', () => {
```
Comment (tkyi) [inline test/plugins/commands.test.js:1198]:
```suggestion
        it('returns 403 when is pull request, () => {
```

Comment (ibu1224) [reply test/plugins/commands.test.js:1198]: @tkyi  thank you

**Review by tk3fftk — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**